org.eclipse.languageserver/src/org/eclipse/languageserver/operations/codelens/LSPCodeLensMenu.java

```diff
@@ -41,45 +33,27 @@
 import io.typefox.lsapi.CodeLens;
 import io.typefox.lsapi.CodeLensParams;
 import io.typefox.lsapi.builders.CodeLensParamsBuilder;
-import io.typefox.lsapi.services.transport.client.LanguageClientEndpoint;
 
 public class LSPCodeLensMenu extends ContributionItem implements IWorkbenchContribution {
 
-	private LanguageClientEndpoint languageClient;
-	private URI fileUri;
+	private LSPDocumentInfo info;
 
 	@Override
 	public void initialize(IServiceLocator serviceLocator) {
 		IEditorPart editor = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActiveEditor();
 		if (editor instanceof ITextEditor) {
-			IEditorInput input = editor.getEditorInput();
-			languageClient = null;
-			fileUri = null;
-			try {
-				IDocument document = null;
-				if (input instanceof IFileEditorInput) { // TODO, also support non resource file
-					IFile file = ((IFileEditorInput) input).getFile();
-					fileUri = file.getLocation().toFile().toURI();
-					document = ITextFileBufferManager.DEFAULT.getTextFileBuffer(file.getFullPath(),	LocationKind.IFILE).getDocument();
-					languageClient = LanguageServiceAccessor.getLanguageServer(file, document);
-				} else if (input instanceof IURIEditorInput) {
-					fileUri = ((IURIEditorInput)input).getURI();
-					document = ITextFileBufferManager.DEFAULT.getTextFileBuffer(new Path(fileUri.getPath()), LocationKind.LOCATION).getDocument();
-					// TODO server
-				}
-			} catch (Exception e) {
-				// TODO: handle exception
-			}
+			info = LanguageServiceAccessor.getLSPDocumentInfoFor((ITextEditor) editor, null);
```

Comment:
the predicate should be `capabilities -> { capabilites.getCodelensProvider() != null }` rather than `null`.